fed/mosaic-sns/src/main/java/org/eclipse/mosaic/fed/sns/ambassador/TransmissionSimulator.java

```diff
@@ -92,21 +93,21 @@ public TransmissionSimulator(RandomNumberGenerator randomNumberGenerator, CSns c
         Map<String, TransmissionResult> transmissionResults = null;
         switch (dac.getType()) {
             case AD_HOC_TOPOCAST:
-                log.debug(
-                        "Send v2xMessage.id={} from node={} as Topocast (singlehop) @time={}",
-                        interaction.getMessage().getId(),
-                        senderName,
-                        TIME.format(interaction.getTime())
-                );
+                if (log.isDebugEnabled()) {
+                    log.debug(
+                            "Send v2xMessage.id={} from node={} as Topocast (singlehop) @time={}",
+                            interaction.getMessage().getId(), senderName, TIME.format(interaction.getTime())
+                    );
+                }
                 transmissionResults = sendMessageAsTopocast(senderName, dac);
                 break;
             case AD_HOC_GEOCAST:
-                log.debug(
-                        "Send v2xMessage.id={} from={} as Geocast (geo routing) @time={}",
-                        interaction.getMessage().getId(),
-                        senderName,
-                        TIME.format(interaction.getTime())
-                );
+                if (log.isDebugEnabled()) {
+                    log.debug(
+                            "Send v2xMessage.id={} from={} as Geocast (geo routing) @time={}",
+                            interaction.getMessage().getId(), senderName, TIME.format(interaction.getTime())
+                    );
+                }
```

Comment:
Using `TIME.format` everytime is rather expensive. A check for `isDebugEnabled` has been added.